### PR TITLE
auth: update design pattern

### PR DESCRIPTION
**Pull request by ayocodingit**
# overview

- separate handler with http
- auth: update design pattern

cc: @jabardigitalservice/jds-backend @firmanJS @sandisunandar99 

## Evidence
- title: auth: update design pattern
- project: Desa Digital
- participants: @ayocodingit @firmanJS @tukangremot @rizkyrmsyah @sandisunandar99

### CODE REVIEW

**Review by firmanJS — APPROVED** [review]

**Review by tukangremot — APPROVED** [review]

File: src/modules/auth/auth_http.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import config from '../../config'
+
+const apiLimiterSignIn = rateLimit({
+  windowMs: Number(config.get('api.limiter.time.signin', 300000)), // Default time signin 5 minutes
+  max: Number(config.get('api.limiter.max.signin', 3)),
+});
+
+const router = Router()
+
+router.post('/v1/auth/users/sign-up', validate(Rules.signUp), Handler.signUp)
```
Comment (sandisunandar99) [inline src/modules/auth/auth_http.ts:16]: better ini `'/v1/auth/users/` kalo bisa pake router group / prefix sih jadi lebih simpel

**Review by rizkyrmsyah — APPROVED** [review]

File: src/modules/auth/auth_http.ts (hunk 1 above)
Comment (ayocodingit) [reply src/modules/auth/auth_http.ts:16]: bisa aja mang, berarti di tambahinnya saat diuse di server.ts, 
jadi seperti ini, `this.app.use('/v1/auth/users', auth)`
image

Comment (ayocodingit) [reply src/modules/auth/auth_http.ts:16]: https://expressjs.com/en/4x/api.html#router

Comment (sandisunandar99) [reply src/modules/auth/auth_http.ts:16]: oh yaya,, is oke atuh,, bener sih gitu aja kalo ingin jadi dinamic toute mah 👍

**Pull request merged**